openpype/modules/kitsu/plugins/publish/integrate_kitsu_review.py

```diff
@@ -29,10 +30,46 @@ def process(self, instance):
             # Skip if not tagged as review
             if "kitsureview" not in representation.get("tags", []):
                 continue
+
+            files_idx = representation.get("files")
+
+            custom_frames = instance.data.get("customFrames", [])
+            keep_frame_index = instance.data.get("keepFrameIndex", False)
+            frame_start = instance.data["frameStart"]
+            frame_end = instance.data["frameEnd"]
+
+            # If only one frame force a list
+            if not isinstance(files_idx, list):
+                files_idx = [files_idx]
+
+            extension = representation.get("ext")
+            if not extension:
+                self.log.warning("No extension found in representation.")
+                raise IndexError
+
             review_path = representation.get("published_path")
+            if not review_path:
+                self.log.warning("No publish path found in representation.")
+                raise IndexError
             self.log.debug("Found review at: {}".format(review_path))
-            gazu.task.add_preview(
-                task_id, comment_id, review_path,
-                normalize_movie=getattr(self, 'normalize', True)
-            )
+
+            if "burnin" in representation.get("tags", []):
+                enumerate_files = zip(list(range(frame_start, frame_end+1)), files_idx)
+                # Update the enumerate_files index if custom_frames is given
+                if custom_frames:
+                    enumerate_files = zip(custom_frames, files_idx)
+                files_idx = ["{:04d}.{}".format(index, extension) for index, file in enumerate_files]
+
+            for file_idx in files_idx:
+                image_filepath = self._rename_output_filepath(review_path, extension, file_idx)
+                self.log.info(image_filepath)
+
+                gazu.task.add_preview(
+                    task_id, comment_id, image_filepath, normalize_movie=getattr(self, 'normalize', True)
+                )
+
             self.log.info("Review upload on comment")
+
+    def _rename_output_filepath(self, published_path, extension, file_idx):
```

Comment:
Like my previous comment I would use `file_index` instead of `file_idx`

Reply:
yes